app/SelectManagers/Product.php

```diff
@@ -566,6 +566,17 @@ protected function boolFilterLinkedWithProductFamily(array &$result)
         ];
     }
 
+    protected function boolFilterWithoutMainImage(&$result)
+    {
+        $query = "SELECT id FROM product WHERE JSON_EXTRACT(data, '$.mainImages') IS NULL OR JSON_LENGTH(JSON_EXTRACT(data, '$.mainImages')) = 0";
```

Comment:
What is it????

Which asset is main image you can understand from product_asset table